drivers/dma/dma_arc_hs.c

```diff
@@ -810,16 +616,103 @@ int dma_arc_hs_transfer(const struct device *dev, uint32_t channel, const void *
 	return -ETIMEDOUT;
 }
 
+static void dma_arc_hs_process_handle(const struct device *dev, struct arc_dma_data *data,
+				      uint32_t handle)
+{
+	const struct arc_dma_config *config = dev->config;
+
+	/* Find which channel this handle belongs to */
+	for (int ch = 0; ch < config->channels; ch++) {
+		struct arc_dma_channel *chan = &data->channels[ch];
+
+		if (chan->state != ARC_DMA_ACTIVE || chan->handle != handle) {
+			continue;
+		}
+
+		chan->state = ARC_DMA_IDLE;
+
+		/* Dispatch callback */
+		/* TODO: Implement callback handling using semaphore */
+		if (chan->callback != NULL) {
+			chan->callback(dev, chan->callback_arg, ch, 0);
+		}
+
+		/* Handle channel linking */
+		if (chan->config.source_chaining_en != 0U || chan->config.dest_chaining_en != 0U) {
+			uint32_t linked_ch = chan->config.linked_channel;
+
+			if (linked_ch < config->channels) {
+				dma_arc_hs_trigger_linked_channel(dev, data, chan, linked_ch);
+			}
+		}
+
+		break;
+	}
+}
+
+static void dma_arc_hs_isr(const struct device *dev)
+{
+	struct arc_dma_data *data = dev->data;
+	uint32_t int_status;
+	uint32_t bits_to_clear = 0;
+
+	while (true) {
+		int_status = z_arc_v2_aux_reg_read(DMA_C_INTSTAT_AUX);
+
+		if (int_status) {
+			/* Read current done status for group 0 */
+			if (int_status & DMA_C_INTSTAT_DONE) {
+				bits_to_clear = z_arc_v2_aux_reg_read(DMA_S_DONESTATD_AUX(0));
+			} else {
+				bits_to_clear = 0;
+			}
+
+			/* clear the interrupt */
+			z_arc_v2_aux_reg_write(DMA_C_INTSTAT_CLR_AUX, int_status);
+
+			if (bits_to_clear) {
+				/* clear the done status */
+				z_arc_v2_aux_reg_write(DMA_S_DONESTATD_CLR_AUX(0), bits_to_clear);
+			}
+		} else {
+			break;
+		}
+
+		/* Handle bus error */
+		if (int_status & DMA_C_INTSTAT_BUS_ERR) {
+			LOG_ERR("DMA bus error");
+			/* TODO: Implement error handling - abort active channels */
+		}
+
+		/* Handle overflow */
+		if (int_status & DMA_C_INTSTAT_OVERFLOW) {
+			LOG_ERR("DMA overflow");
+			/* TODO: Implement overflow handling */
+		}
+
+		if ((int_status & DMA_C_INTSTAT_DONE) && (bits_to_clear != 0)) {
```

Comment:
Similar thing about implicit conversion
```suggestion
		if (((int_status & DMA_C_INTSTAT_DONE) != 0) && (bits_to_clear != 0)) {
```